File: Desktop/src/Entities/Panier.java

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package Entities;

import Utils.Enumerations;
import Utils.Enumerations.ModeLivraison;
import Utils.Enumerations.ModePaiement;
import Utils.Enumerations.StatusPanier;
import java.io.IOException;
import java.sql.Date;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Objects;
import org.apache.pdfbox.pdmodel.PDDocument;
import org.apache.pdfbox.pdmodel.PDPage;
import org.apache.pdfbox.pdmodel.PDPageContentStream;
import org.apache.pdfbox.pdmodel.font.PDFont;
import org.apache.pdfbox.pdmodel.font.PDType1Font;

/**
 *
 * @author monta
 */
public class Panier {

    private int id;
    private User user;
    private LocalDateTime dateCreation;
    private LocalDateTime dateLivraison;
    private double totalTTC;
    private double fraisLivraison;
    private StatusPanier status;
    private ModePaiement modePaiement;
    private ModeLivraison modeLivraison;
    private boolean estLivre;
    private boolean estPaye;
    private List<ProduitPanier> contenu;

    public Panier(int id, User user, LocalDateTime dateCreation, LocalDateTime dateLivraison, double totalTTC, double fraisLivraison, String status, String modePaiement, String modeLivraison, boolean estLivre, boolean estPaye, List<ProduitPanier> contenu) {
        this.id = id;
        this.user = user;
        this.dateCreation = dateCreation;
        this.dateLivraison = dateLivraison;
        this.totalTTC = totalTTC;
        this.fraisLivraison = fraisLivraison;
        this.status = StatusPanier.valueOf(status);
        this.modePaiement = ModePaiement.valueOf(modePaiement);
        this.estLivre = estLivre;
        this.estPaye = estPaye;
        this.contenu = contenu;
        this.modeLivraison = ModeLivraison.valueOf(modeLivraison.replace(" ", ""));
    }

    public Panier(User user, LocalDateTime dateCreation) {
        this.user = user;
        this.dateCreation = dateCreation;
        this.status = StatusPanier.Temporelle;
        this.modeLivraison = ModeLivraison.Surplace;

    }

    public Panier() {
    }

    public ModeLivraison getModeLivraison() {
        return modeLivraison;
    }

    public void setModeLivraison(ModeLivraison modeLivraison) {
        this.modeLivraison = modeLivraison;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public LocalDateTime getDateCreation() {
        return dateCreation;
    }

    public void setDateCreation(LocalDateTime dateCreation) {
        this.dateCreation = dateCreation;
    }

    public LocalDateTime getDateLivraison() {
        return dateLivraison;
    }

    public void setDateLivraison(LocalDateTime dateLivraison) {
        this.dateLivraison = dateLivraison;
    }

    public double getTotalTTC() {
        return totalTTC;
    }

    public void setTotalTTC(double totalTTC) {
        this.totalTTC = totalTTC;
    }

    public double getFraisLivraison() {
        return fraisLivraison;
    }

    public void setFraisLivraison(double fraisLivraison) {
        this.fraisLivraison = fraisLivraison;
    }

    public StatusPanier getStatus() {
        return status;
    }

    public void setStatus(StatusPanier status) {
        this.status = status;
    }

    public ModePaiement getModePaiement() {
        return modePaiement;
    }

    public void setModePaiement(ModePaiement modePaiement) {
        this.modePaiement = modePaiement;
    }

    public boolean isEstLivre() {
        return estLivre;
    }

    public void setEstLivre(boolean estLivre) {
        this.estLivre = estLivre;
    }

    public boolean isEstPaye() {
        return estPaye;
    }

    public void setEstPaye(boolean estPaye) {
        this.estPaye = estPaye;
    }

    public List<ProduitPanier> getContenu() {
        return contenu;
    }

    public void setContenu(List<ProduitPanier> contenu) {
        this.contenu = contenu;
    }

    @Override
    public String toString() {
        return "Panier{" + "id=" + id + ", userId=" + user.getId() + ", dateCreation=" + dateCreation + ", dateLivraison=" + dateLivraison + ", totalTTC=" + totalTTC + ", fraisLivraison=" + fraisLivraison + ", status=" + status + ", modePaiement=" + modePaiement + ", estLivre=" + estLivre + ", estPaye=" + estPaye + ", contenu=" + contenu + '}';
    }

    @Override
    public int hashCode() {
        int hash = 3;
        hash = 31 * hash + Objects.hashCode(this.id);
        hash = 31 * hash + Objects.hashCode(this.user.getId());
        hash = 31 * hash + Objects.hashCode(this.dateCreation);
        hash = 31 * hash + Objects.hashCode(this.dateLivraison);
        hash = 31 * hash + (int) (Double.doubleToLongBits(this.totalTTC) ^ (Double.doubleToLongBits(this.totalTTC) >>> 32));
        hash = 31 * hash + (int) (Double.doubleToLongBits(this.fraisLivraison) ^ (Double.doubleToLongBits(this.fraisLivraison) >>> 32));
        hash = 31 * hash + Objects.hashCode(this.status);
        hash = 31 * hash + Objects.hashCode(this.modePaiement);
        hash = 31 * hash + Objects.hashCode(this.contenu);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Panier other = (Panier) obj;
        if (Double.doubleToLongBits(this.totalTTC) != Double.doubleToLongBits(other.totalTTC)) {
            return false;
        }
        if (Double.doubleToLongBits(this.fraisLivraison) != Double.doubleToLongBits(other.fraisLivraison)) {
            return false;
        }
        if (!Objects.equals(this.id, other.id)) {
            return false;
        }
        if (!Objects.equals(this.user.getId(), other.user.getId())) {
            return false;
        }
        if (!Objects.equals(this.status, other.status)) {
            return false;
        }
        if (!Objects.equals(this.modePaiement, other.modePaiement)) {
            return false;
        }
        if (!Objects.equals(this.dateCreation, other.dateCreation)) {
            return false;
        }
        if (!Objects.equals(this.dateLivraison, other.dateLivraison)) {
            return false;
        }
        if (!Objects.equals(this.contenu, other.contenu)) {
            return false;
        }
        return true;
    }

    public String genererMailBody() {
        String body = "";
        for (ProduitPanier p : this.contenu) {
            body += "<tr><td>" + p.reference + "</td>" + "<td>" + p.libelle + "</td>" + "<td>" + p.getQuantiteVendue() + "</td>" + "<td>" + p.getPrixVente() * p.getQuantiteVendue() + "</td></tr>";
        }
        return body;
    }

    public boolean generatePDF() {
        try {
            PDDocument myPdf = new PDDocument();
            PDPage page = new PDPage();
            myPdf.addPage(page);
            PDPageContentStream contentStream = new PDPageContentStream(myPdf, page);
            contentStream.beginText();
            contentStream.setFont(PDType1Font.TIMES_ROMAN, 36);
            contentStream.setLeading(14.5f);

            contentStream.newLineAtOffset(25, 700);
            contentStream.showText("                       Commande");
            contentStream.newLine();
            contentStream.newLine();
            contentStream.newLine();
            contentStream.showText(this.dateCreation.toString().replace("T", " "));
            contentStream.newLine();
            contentStream.newLine();
            contentStream.newLine();
            contentStream.setFont(PDType1Font.TIMES_ROMAN, 14);
            contentStream.showText("Au nom de :" + this.user.getNom() + " " + this.user.getPrenom());
            contentStream.newLine();
            contentStream.newLine();
            contentStream.newLine();
            contentStream.showText("_______________________________________________________________________________________________________________");

            for (ProduitPanier p : this.contenu) {
                contentStream.newLine();
                contentStream.newLine();
                contentStream.showText("Reference : " + p.reference + " Libelle :" + p.libelle + " quantité:" + p.getQuantiteVendue() + " prix :" + p.getPrixVente() + " Total :" + p.getPrixVente() * p.getQuantiteVendue() + "  Boutique : " + p.boutique.getNom());
                contentStream.newLine();
                contentStream.showText("_______________________________________________________________________________________________________________");
            }
            contentStream.endText();
            contentStream.close();
            myPdf.save(System.getProperty("user.home") + "/Desktop"+ this.id + ".pdf");
            myPdf.close();
        } catch (IOException e) {
            System.out.println(e);
            return false;
        }
        return true;
    }
}

File: WEB/src/SoukBundle/Entity/Panier.php
<?php

namespace SoukBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Panier
 *
 * @ORM\Table(name="panier", indexes={@ORM\Index(name="id_user", columns={"id_user"})})
 * @ORM\Entity
 */
class Panier
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_creation", type="datetime", nullable=false)
     */
    private $dateCreation;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_livraison", type="datetime", nullable=true)
     */
    private $dateLivraison;

    /**
     * @var string
     *
     * @ORM\Column(name="total_ttc", type="decimal", precision=7, scale=3, nullable=false)
     */
    private $totalTtc;

    /**
     * @var string
     *
     * @ORM\Column(name="frais_livraison", type="decimal", precision=7, scale=0, nullable=false)
     */
    private $fraisLivraison;

    /**
     * @var string
     *
     * @ORM\Column(name="statut", type="string", length=20, nullable=false)
     */
    private $statut;

    /**
     * @var string
     *
     * @ORM\Column(name="mode_paiement", type="string", length=20, nullable=false)
     */
    private $modePaiement;

    /**
     * @var string
     *
     * @ORM\Column(name="mode_livraison", type="string", length=20, nullable=true)
     */
    private $modeLivraison;

    /**
     * @var boolean
     *
     * @ORM\Column(name="est_livre", type="boolean", nullable=false)
     */
    private $estLivre;

    /**
     * @var boolean
     *
     * @ORM\Column(name="est_paye", type="boolean", nullable=false)
     */
    private $estPaye;

    /**
     * @var User
     *
     * @ORM\ManyToOne(targetEntity="User")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="id_user", referencedColumnName="id")
     * })
     */
    private $idUser;



    /**
     * Get id
     *
     * @return integer
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set dateCreation
     *
     * @param \DateTime $dateCreation
     *
     * @return Panier
     */
    public function setDateCreation($dateCreation)
    {
        $this->dateCreation = $dateCreation;

        return $this;
    }

    /**
     * Get dateCreation
     *
     * @return \DateTime
     */
    public function getDateCreation()
    {
        return $this->dateCreation;
    }

    /**
     * Set dateLivraison
     *
     * @param \DateTime $dateLivraison
     *
     * @return Panier
     */
    public function setDateLivraison($dateLivraison)
    {
        $this->dateLivraison = $dateLivraison;

        return $this;
    }

    /**
     * Get dateLivraison
     *
     * @return \DateTime
     */
    public function getDateLivraison()
    {
        return $this->dateLivraison;
    }

    /**
     * Set totalTtc
     *
     * @param string $totalTtc
     *
     * @return Panier
     */
    public function setTotalTtc($totalTtc)
    {
        $this->totalTtc = $totalTtc;

        return $this;
    }

    /**
     * Get totalTtc
     *
     * @return string
     */
    public function getTotalTtc()
    {
        return $this->totalTtc;
    }

    /**
     * Set fraisLivraison
     *
     * @param string $fraisLivraison
     *
     * @return Panier
     */
    public function setFraisLivraison($fraisLivraison)
    {
        $this->fraisLivraison = $fraisLivraison;

        return $this;
    }

    /**
     * Get fraisLivraison
     *
     * @return string
     */
    public function getFraisLivraison()
    {
        return $this->fraisLivraison;
    }

    /**
     * Set statut
     *
     * @param string $statut
     *
     * @return Panier
     */
    public function setStatut($statut)
    {
        $this->statut = $statut;

        return $this;
    }

    /**
     * Get statut
     *
     * @return string
     */
    public function getStatut()
    {
        return $this->statut;
    }

    /**
     * Set modePaiement
     *
     * @param string $modePaiement
     *
     * @return Panier
     */
    public function setModePaiement($modePaiement)
    {
        $this->modePaiement = $modePaiement;

        return $this;
    }

    /**
     * Get modePaiement
     *
     * @return string
     */
    public function getModePaiement()
    {
        return $this->modePaiement;
    }

    /**
     * Set modeLivraison
     *
     * @param string $modeLivraison
     *
     * @return Panier
     */
    public function setModeLivraison($modeLivraison)
    {
        $this->modeLivraison = $modeLivraison;

        return $this;
    }

    /**
     * Get modeLivraison
     *
     * @return string
     */
    public function getModeLivraison()
    {
        return $this->modeLivraison;
    }

    /**
     * Set estLivre
     *
     * @param boolean $estLivre
     *
     * @return Panier
     */
    public function setEstLivre($estLivre)
    {
        $this->estLivre = $estLivre;

        return $this;
    }

    /**
     * Get estLivre
     *
     * @return boolean
     */
    public function getEstLivre()
    {
        return $this->estLivre;
    }

    /**
     * Set estPaye
     *
     * @param boolean $estPaye
     *
     * @return Panier
     */
    public function setEstPaye($estPaye)
    {
        $this->estPaye = $estPaye;

        return $this;
    }

    /**
     * Get estPaye
     *
     * @return boolean
     */
    public function getEstPaye()
    {
        return $this->estPaye;
    }

    /**
     * Set idUser
     *
     * @param User $idUser
     *
     * @return Panier
     */
    public function setIdUser(User $idUser = null)
    {
        $this->idUser = $idUser;

        return $this;
    }

    /**
     * Get idUser
     *
     * @return User
     */
    public function getIdUser()
    {
        return $this->idUser;
    }
}